app/page/info/tokens.tsx

```diff
@@ -0,0 +1,229 @@
+        Tokens:
+        {tokens &&
+          tokens.map((token, index) => {
+            return (
+              <div key={index}>
+                <a
+                  href={`${explorerUrl}token/${token.address}`}
+                  target="_blank">
+                  {`${token.symbol}`}
+                </a>
+                {` ${parseFloat(
+                  ethers.formatUnits(
+                    ethers.toBeHex(token.balance),
+                    ethers.toNumber(token.decimals)
+                  )
+                ).toFixed(6)}`}
+              </div>
+            )
+          })}
+      </div>
+      <div className="col-span-3 cursor-pointer" onClick={toggleAccountModal}>
+        <span>Import Tokens</span>
+        <FontAwesomeIcon icon={faCaretDown} className="ml-2" />
+      </div>
+      {isTokenModalOpened && <TokenModal onModalClosed={toggleAccountModal} />}
+    </div>
+  )
+}
+
+function TokenModal({ onModalClosed }: { onModalClosed: () => void }) {
+  const { provider } = useProviderContext()
+  const { importToken } = useAction()
+  const account = useAccount()
+
+  const [tokenAddress, setTokenAddress] = useState<HexString>("")
+  const [tokenName, setTokenName] = useState<string>("")
+  const [tokenSymbol, setTokenSymbol] = useState<string>("")
+  const [tokenDecimals, setTokenDecimals] = useState<string>("")
+  const [firstOpenTokenModal, setFirstOpenTokenModal] = useState<boolean>(true)
+  const [invalidTokenAddress, setInvalidTokenAddress] = useState<boolean>(true)
+  const [invalidTokenSymbol, setInvalidTokenSymbol] = useState<boolean>(false)
+  const [invalidTokenDecimals, setInvalidTokenDecimals] =
+    useState<boolean>(false)
+
+  const handleTokenAddressChange = async (
+    event: ChangeEvent<HTMLInputElement>
+  ) => {
+    setFirstOpenTokenModal(false)
+    const inputTokenAddress = event.target.value
+    setTokenAddress(inputTokenAddress)
+    const erc20 = getErc20Contract(inputTokenAddress, provider)
+
+    try {
+      console.log(`${ethers.getAddress(inputTokenAddress)}`)
+      const name: string = await erc20.name()
+      const symbol: string = await erc20.symbol()
+      const decimals: number = ethers.toNumber(await erc20.decimals())
+      setInvalidTokenAddress(false)
+      setTokenName(name)
+      setTokenSymbol(symbol)
+      setTokenDecimals(decimals.toString())
+    } catch (error) {
+      console.warn(`[Popup][tokens] Invalid token address: ${error}`)
+      setInvalidTokenAddress(true)
+      setTokenName("")
+      setTokenSymbol("")
+      setTokenDecimals("")
+    }
+  }
+
+  const handleTokenSymbolChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const inputTokenSymbol = event.target.value
+    setTokenSymbol(inputTokenSymbol)
+    setInvalidTokenSymbol(inputTokenSymbol.length === 0)
+  }
+
+  const handleTokenDecimalsChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const inputTokenDecimals = event.target.value
```

Comment:
As we discussed last time, we should not allow user to change decimals manually.